modules/monitoring-supported-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,314 @@
+
+|TLS client|`tlsConfig`|A TLS client configuration specifies the CA certificate, the client certificate, and the client key file information used to authenticate with the remote write endpoint server using TLS.
+The sample configuration assumes that you have already created a CA certificate file, a client certificate file, and a client key file.
+
+|===
+
+== Config map location for authentication settings
+The following shows the location of the authentication configuration in the `ConfigMap` object for default platform monitoring.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write-endpoint.example.com" <1>
+        <endpoint_authentication_details> <2>
+----
+<1> The URL of the remote write endpoint.
+<2> The required configuration details for the authentication method for the endpoint.
+Currently supported authentication methods are Amazon Web Services (AWS) Signature Version 4, bearer token, basic authentication, OAuth 2.0, and TLS client. 
+
+[NOTE]
+====
+If you want to configure remote write for the Prometheus instance that monitors user-defined projects, edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace.
+====
+
+== Example remote write authentication settings
+
+The following samples show different authentication settings you can use to connect to a remote write endpoint.
+Each sample also shows how to configure a corresponding `Secret` object that contains authentication credentials and other relevant settings.
+Each sample configures authentication for use with default platform monitoring in the `openshift-monitoring` namespace.
+
+.Sample YAML for AWS Signature Version 4 authentication
+
+The following shows the settings for a `sigv4` secret named `sigv4-credentials` in the `openshift-monitoring` namespace.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: sigv4-credentials
+  namespace: openshift-monitoring
+stringData:
+  access: <AWS_access_key> <1>
+  secret: <AWS_secret_key> <2>
+type: Opaque
+----
+<1> The AWS API access key.
+<2> The AWS API secret key.
+
+The following shows sample AWS Signature Version 4 remote write authentication settings that use a `Secret` object named `sigv4-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        sigv4:
+          region: <AWS_region> <1>
+          accessKey:
+            name: sigv4-credentials <2>
+            key: access <3>
+          secretKey:
+            name: sigv4-credentials <2>
+            key: secret <3>
```

Comment:
Should we represent any of the values in the preceding five lines be between angle brackets?

Reply:
No. These are the actual values. They correspond to the fields in the `Secret` object that contain the credentials.